### ISSUE TITLE
SaveLoad does not work for files on Amazon S3 (propagated bug from smart_open)

### ISSUE THREAD

**Issue by ziky90** [piskvorky/gensim#421]
I am not sure if this should have been reported in here or in smart_open.

There is not working `SaveLoad` for objects stored on S3.
I remember that this used to work, but probably due to some changes in cPickle it does not work now.

I am particularly using this as:
```
model = MyModel.load(model_path)
``` 
When I call it on the same file locally, then it works, when I call it from S3, then it stops working with stack trace:
```
/usr/local/lib/python2.7/dist-packages/gensim/utils.pyc in load(cls, fname, mmap)
    250         compress, subname = SaveLoad._adapt_by_suffix(fname)
    251
--> 252         obj = unpickle(fname)
    253         obj._load_specials(fname, mmap, compress, subname)
    254         return obj

/usr/local/lib/python2.7/dist-packages/gensim/utils.pyc in unpickle(fname)
    905     """Load pickled object from `fname`"""
    906     with smart_open(fname) as f:
--> 907         return _pickle.load(f)
    908
    909

TypeError: argument must have 'read' and 'readline' attributes
```

I have further investigated that:
```
with smart_open("s3://mybucket/model.pkl") as f:
    print pickle.load(f)
```
ends again with `TypeError: argument must have 'read' and 'readline' attributes`

But when I do:
```
str = ""
for line in smart_open("s3://mybucket/model.pkl"):
    str = str + line
pickle.loads(str)
```
Everything works fine.

**The problem can be solved by:**

1) Workaround with conversion to string and calling `loads`?
2) `readline` implementation in `S3OpenRead`? (I guess preferable)

**Comment by piskvorky** [comment]
Yes, both options seem fine. The "string 1) option" can also be written more simply as `pickle.loads(smart_open(f).read())`.

Adding `readline()` is preferable, yes. But I don't remember how exactly the protocol works, it looks a little tricky. How is `readline` supposed to interact with `read()`, buffering etc.

**Comment by piskvorky** [comment]
Fixed in #422. Thanks @ziky90 !
